docs/the-new-architecture/cxx-custom-types.md

```diff
@@ -0,0 +1,284 @@
+
+This guide continues the previous [C++ Turbo Native Modules](./cxx-cxxturbomodules) section.
+
+## Example: Int64
+
+C++ Turbo Native Modules don't support `int64_t` numbers yet - because JavaScript doesn't support numbers greater `2^53`.
+
+We can't represent numbers > `2^53` as JavaScript `number`'s - but we can represent them as JavaScript `string`'s and automatically convert (aka. `bridge`) them to C++ `int64_t`'s by creating a custom Bridging header file in the `tm` folder:
+
+```cpp Int64.h
+#pragma once
+
+#include <react/bridging/Bridging.h>
+
+namespace facebook::react {
+
+template <>
+struct Bridging<int64_t> {
+  static int64_t fromJs(jsi::Runtime &rt, const jsi::String &value) {
+    try {
+      size_t pos;
+      auto str = value.utf8(rt);
+      auto num = std::stoll(str, &pos);
+      if (pos != str.size()) {
+        throw std::invalid_argument("Invalid number"); // don't support alphanumeric strings
+      }
+      return num;
+    } catch (const std::logic_error &e) {
+      throw jsi::JSError(rt, e.what());
+    }
+  }
+
+  static jsi::String toJs(jsi::Runtime &rt, int64_t value) {
+    return bridging::toJs(rt, std::to_string(value));
+  }
+};
+
+} // namespace facebook::react
+```
+
+The key components for your custom `bridging` header are:
+- Explicit specialization of the `Bridging` struct for your custom type, `int64_t` in this case
+- A `fromJs` function to convert from `jsi::` types to your custom type
+- A `toJS` function to convert from your custom type to `jsi:` types
+
+Omitting either `fromJS` or `toJS` would make you `bridging` header either *readonly* or *writeonly*.
+
+Now you can add the following function to your JavaScript spec:
+
+<Tabs groupId="turbomodule-specs" defaultValue={constants.defaultJavaScriptSpecLanguages} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+```typescript title="NativeSampleModule.ts"
+   // ...
+   readonly passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+<TabItem value="flow">
+
+```js title="NativeSampleModule.js"
+   // ...
+   +passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+</Tabs>
+
+Declare it in your `NativeSampleModule.h` file and include the `Int64.h` header file:
+
+```cpp
+//...
+#include "Int64.h"
+//...
+int32_t passLargeNumber(jsi::Runtime& rt, int64_t input);
+```
+
+And implement it in `NativeSampleModule.cpp`:
+
+```cpp
+//...
+#include <cmath>
+//...
+int32_t NativeSampleModule::passLargeNumber(jsi::Runtime& rt, int64_t input) {
+    return std::cbrt(input);
+}
+```
+
+In your app you can call this new native function via:
+
+```js
+<Section title="Cxx TurboModule">
+  NativeSampleModule.passLargeNumber(...) ={' '}
+  {JSON.stringify(
+    NativeSampleModule.passLargeNumber('9223372036854775807')
+  )}
+</Section>
+```
+
+which should return `2097152`.
```

Comment:
It's not clear why `passLargeNumber` should return the cubic root and from where this value come from. What about renaming `passLargeNumber` to `cubicRoot`? I think it will make it easier to follow the example, wdyt?

Reply:
That makes much more sense!
Just needed to turn the largest int64 into an int32